File: GoCoder/answer/1_two_sum.go
package answer

func TwoSum(nums []int, target int)[] int{
	hashmap := map[int]int{}
	for k, v := range nums {
		if p, ok := hashmap[target-v]; ok{
			return []int{p, k}
		}
		hashmap[v] = k
	}
	return nil
}

File: PyCoder/answer/1_two_sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        hashdict = {}
        for i in range(len(nums)):
            if (target-nums[i]) in hashdict:
                return [i, hashdict[target-nums[i]]]
            hashdict[nums[i]] = i
        return []


if __name__ == "__main__":
    nums = [2, 7, 11, 15]
    target = 9
    s = Solution()
    print(s.twoSum(nums, target))